docs/docs/typescript/generated-types.md

```diff
@@ -0,0 +1,157 @@
+---
+description: A look at automatic type generation in Redwood
+---
+
+# Generated Types
+
+To add to the TypeScript (and JavaScript!) experience, Redwood generates types for you.
+These generated types not only include your GraphQL operations, but also your named routes, Cells, scenarios, and tests.
+
+When you run `yarn rw dev`, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too:
+
+```
+yarn rw g types
+```
+
+:::tip Getting errors trying to generate types?
+
+If you're getting errors trying to generate types, it's worth checking the GraphQL operations in your Cells and SDLs.
+Make sure that they're syntactically valid, and that every query and mutation on the web side is defined in an `*.sdl.js` file on the api side.
+
+:::
+
+If you're curious, you can find the generated types in the `.redwood/types`, `web/types/graphql.d.ts`, and `api/types/graphql.d.ts` directories. Broadly speaking, Redwood generates the following types:
+
+1. ["mirror" types](https://www.typescriptlang.org/docs/handbook/module-resolution.html#virtual-directories-with-rootdirs) for your components, Cells, and layouts on the web side
```

Comment:
Is there more than what's enumerated here? Maybe the "mirror" types include the api side too? For a simple project with a Post model and dbAuth setup, this is what I'm seeing:

```
.redwood/types/mirror/
├── web/
│  └── src/
│     └── pages/
│        ├── NotFoundPage/
│        │  └── index.d.ts
│        └── FatalErrorPage/
│           └── index.d.ts
└── api/
   └── src/
      ├── services/
      │  └── posts/
      │     └── index.d.ts
      └── directives/
         ├── skipAuth/
         │  └── index.d.ts
         └── requireAuth/
            └── index.d.ts
```